ENG-6237: Update README to include streaming examples (#148)
fauna/fauna-go: README.md

}
```

## Streaming

> Streams can be initiated with the `Client.Stream()` method:

Reviewer: ```suggestion
To start and subscribe to a stream, pass a query that produces a stream token to `Client.Stream()`:
```

``` go
package main

import (
	"fmt"

	"github.com/fauna/fauna-go"
)

type Product struct {
	Name     string  `fauna:"name"`
	Category string  `fauna:"category"`
	Price    float64 `fauna:"price"`
}

func main() {
	client, err := fauna.NewDefaultClient()
	if err != nil {
		panic(err)
	}

	streamQuery, _ := fauna.FQL("Product.all().toStream()", nil)
	events, err := client.Stream(streamQuery)
	if err != nil {
		panic(err)
	}
	defer events.Close()

	for {
		event, err := events.Next()
		if err != nil {
			panic(err)
		}

		switch event.Type {
		case fauna.AddEvent, fauna.UpdateEvent, fauna.RemoveEvent:
			var product Product
			if err = event.Unmarshal(&product); err != nil {
				panic(err)
			}
			fmt.Println(product)
		}
	}
}
```

> Streams can compose with normal query results. In order to initiate a stream from a query result, call the `Client.Subscribe()` method on the resulting `fauna.Stream` value.

Reviewer: ```suggestion
Streams can compose with normal query results. To start a stream from a query result, call `Client.Subscribe()` on the `fauna.Stream` value.
```

``` go
package main

import (
	"fmt"

	"github.com/fauna/fauna-go"
)

type Product struct {
	Name     string  `fauna:"name"`
	Category string  `fauna:"category"`
	Price    float64 `fauna:"price"`
}

func main() {
	client, err := fauna.NewDefaultClient()
	if err != nil {
		panic(err)
	}

	dataLoad, _ := fauna.FQL(`
		let products = Product.all()
		{
			Products: products.toArray(),
			Stream: products.toStream()
		}
	`, nil)

	data, err := client.Query(dataLoad)
	if err != nil {
		panic(err)
	}

	queryResult := struct {
		Products []Product
		Stream   fauna.Stream
	}{}

	if err := data.Unmarshal(&queryResult); err != nil {
		panic(err)
	}

	fmt.Println("Existing products:")
	for _, product := range queryResult.Products {
		fmt.Println(product)
	}

	events, err := client.Subscribe(queryResult.Stream)
	if err != nil {
		panic(err)
	}
	defer events.Close()

	fmt.Println("Products from streaming:")
	for {
		event, err := events.Next()
		if err != nil {
			panic(err)
		}

		switch event.Type {
		case fauna.AddEvent, fauna.UpdateEvent, fauna.RemoveEvent:
			var product Product
			if err = event.Unmarshal(&product); err != nil {
				panic(err)
			}
			fmt.Println(product)
		}
	}
}
```

> The client tracks its last seen event timestamp. Upon network failures, the client will attempt to resume streaming at the last seen event time (exclusively). When designing a system that tolerates crashes, the last seen event timestamp must be stored by the application and given to the `Client.Subscribe()` method during recovery via the `fauna.StartTime()` function.

Reviewer: ```suggestion
The client tracks its last seen event's timestamp. After a network failure, the client attempts to resume streaming at this timestamp (exclusively). To recover from crashes, your application must store the last seen event's timestamp and pass it to `Client.Subscribe()` using the `fauna.StartTime()` function.
```
